src/middlewared/middlewared/plugins/kerberos.py

```diff
@@ -187,9 +187,8 @@ async def _validate_param_type(self, data):
                     raise CallError(f'[{e}] is not a supported encryption type')
 
         if data['ptype'] == 'cctype':
-            for cctype in ['DIR', 'FILE', 'MEMORY']:
-                if cctype not in data['value']:
-                    raise CallError(f'{data["value"]} is an unsupported cctype')
+            if data['value'] not in ['DIR', 'FILE', 'MEMORY']:
```

Comment:
Few lines above you normalize `data['value']` by calling `.upper()`. Why not do that here?

Reply:
Underlying kerberos behavior for parsing sting parameters is case-sensitive. I expanded the error message to be a little more intuitive. I'm trying to strike a balance here between a total free-for-all for these aux parameters (leading to more broken configs) and having something more user-friendly (encouraging users to change these parameters - leading to more broken configs).